core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -795,11 +801,12 @@ class DAGScheduler(
 
     // First figure out the indexes of partition ids to compute.
     val partitionsToCompute: Seq[Int] = {
-      if (stage.isShuffleMap) {
-        (0 until stage.numPartitions).filter(id => stage.outputLocs(id) == Nil)
-      } else {
-        val job = stage.resultOfJob.get
-        (0 until job.numPartitions).filter(id => !job.finished(id))
+      stage match {
+        case stage: ShuffleMapStage => (0 until stage.numPartitions).filter(id =>
```

Comment:
Sorry one more small style comment here: newline after "=>" (always do a newline unless all case statements for the patch fit on a single line)